### ISSUE TITLE
How to compute reward

### ISSUE THREAD

**Issue by lpq29743** [qyyin/Reinforce4ZP#1]
I can find how the program computes the reward as follows, but it's not easy for me to understand.
```
thre = output_softmax[:,1].data.cpu().numpy()
lu = numpy.clip(numpy.floor(numpy.random.rand(len(thre)) / thre), 1, 0).astype(int)
gold = data["target"]
if float(sum(gold)) == 0 or sum(gold*lu) == 0 or sum(lu) == 0:continue
prec = float(sum(gold*lu))/float(numpy.count_nonzero(lu))
rec = float(sum(gold*lu))/float(numpy.count_nonzero(gold))
sc = 0 if (rec == 0.0 or prec == 0.0) else 2.0/(1.0/prec+1.0/rec)
if sc == 0:continue
rewards = numpy.full((len(lu),2),sc)
pl = lu.tolist()
for i in range(len(pl)):
    np = copy.deepcopy(pl)
    np[i] = 1-np[i]
    if float(sum(gold)) == 0 or sum(gold*np) == 0 or sum(np) == 0:
        nsc = 0.0
    else:
        nprec = float(sum(gold*np))/float(numpy.count_nonzero(np))
        nrec = float(sum(gold*np))/float(numpy.count_nonzero(gold))
        nsc = 0.0 if (nrec == 0.0 or nprec == 0.0) else 2.0/(1.0/nprec+1.0/nrec)
    rewards[i][np[i]] = nsc
maxs = rewards.min(axis=1)[:,numpy.newaxis]
rewards = rewards - maxs
rewards = torch.tensor(-1.0*rewards).type(torch.cuda.FloatTensor)
```
Here are some of my question:
1. What does `gold` stand for?
2. Why are `prec`, `rec`, `sc` negative numbers?
3. ...
